linode/manager
# change: [poc] - Better subnet validation behavior on VPCCreate page

## Description 📝
Previously when working on the VPCCreate page, I'd made two validation versions:
- The first one did not use the CreateVPCSchema validation for subnets (it validated subnets before using the schema) but had the desired UI -- this one got merged in
- The [second one that used the validation schema](https://github.com/coliu-akamai/manager/pull/1) had UI downsides, where you couldn't leave a subnet blank and have it be ignored

In the current reason, errors for specific subnets might sometimes appear as general subnet errors, because of how subnet validation happened... Wasn't the biggest fan of that behavior, so I looked into this some more. That's where this PR comes in -- it uses the CreateVPCSchema validation and processes any subnet specific errors returned by the API, while still having the same UI behavior of the  current version.

## Major Changes 🔄
- The logic for this PR is built off the logic of the [pr I linked above](https://github.com/coliu-akamai/manager/pull/1) -   I basically just added a map of the UI subnet indexes to the indexes of the subnets that get sent to the backend 😅 
- removed the now unused subnet validation logic 

## Preview 📷

| Before  | After   |
| ------- | ------- |
| ![image](https://github.com/linode/manager/assets/139280159/c1e2d82b-2584-4d6b-a00e-17a832a80bfa) | ![image](https://github.com/linode/manager/assets/139280159/0834445e-6f89-46b9-a08e-728a1911b93d) |

## How to test 🧪
`yarn test formikErrorUtils.test.ts`
1. Try creating VPCs with subnets and ensure that subnet-specific errors correspond with the correct subnet, and that general subnet errors still get handled the same
